
We shouldn't need to set that clock divider at all, because we shouldn't be changing the hdmi reference clock rate in the first place.

Fix this!

Signed-off: derek.foreman@collabora.com


diff --speed-large-files --no-dereference --minimal -Naur linux-6.12.5/drivers/gpu/drm/rockchip/dw_hdmi_qp-rockchip.c linux-6.12.5/drivers/gpu/drm/rockchip/dw_hdmi_qp-rockchip.c
--- linux-6.12.5/drivers/gpu/drm/rockchip/dw_hdmi_qp-rockchip.c	2024-12-19 11:59:04.473573175 +0100
+++ linux-6.12.5/drivers/gpu/drm/rockchip/dw_hdmi_qp-rockchip.c	2024-12-19 12:04:11.416914757 +0100
@@ -80,7 +80,6 @@
 	int rate;
 	u32 val = 0;
 
-	clk_set_rate(hdmi->ref_clk, crtc_state->adjusted_mode.clock * 1000);
 	rate = crtc->state->mode.clock * 10;
 	if (conn_state->hdmi.output_bpc == 10 &&
 	    conn_state->hdmi.output_format != HDMI_COLORSPACE_YUV422) {
@@ -124,7 +123,6 @@
 	if (crtc && crtc->state) {
 		rate = drm_hdmi_compute_mode_clock(&crtc->state->adjusted_mode,
 						   8, HDMI_COLORSPACE_RGB);
-		clk_set_rate(hdmi->ref_clk, rate);
 		/*
 		 * FIXME: Temporary workaround to pass pixel clock rate
 		 * to the PHY driver until phy_configure_opts_hdmi
